packages/core/test/new-api/execution/execution-engine.ts

```diff
@@ -237,6 +266,144 @@ describe("execution engine", () => {
       },
     ]);
   });
+
+  describe("with complex arguments", () => {
+    it("should execute deploy when futures are passed as nested arguments", async () => {
+      const moduleDefinition = defineModule("Module1", (m) => {
+        const library1 = m.library("Library1");
+
+        const account1 = m.getAccount(1);
+        const supply = m.getParameter("supply", 1000);
+
+        const contract1 = m.contract(
+          "Contract1",
+          [{ nested: library1, arr: [account1, supply] }],
+          {
+            libraries: {
+              Library1: library1,
+            },
+          }
+        );
+
+        return { library1, contract1 };
+      });
+
+      const constructor = new ModuleConstructor();
+      const module = constructor.construct(moduleDefinition);
+
+      assert.isDefined(module);
+
+      const executionStateMap = {};
+
+      const batches = Batcher.batch(module, {});
+
+      const executionEngine = new ExecutionEngine();
+      const journal = new MemoryJournal();
+      const accounts: string[] = exampleAccounts;
+      const mockTransactionService = setupMockTransactionService();
+      const mockArtifactResolver = setupMockArtifactResolver();
+
+      const result = await executionEngine.execute({
+        batches,
+        module,
+        executionStateMap,
+        accounts,
+        strategy: new BasicExecutionStrategy(),
+        journal,
+        transactionService: mockTransactionService,
+        artifactResolver: mockArtifactResolver,
+        deploymentParameters: {},
+      });
+
+      assert.isDefined(result);
+      const journalMessages = await accumulateMessages(journal);
+
+      assert.deepStrictEqual(journalMessages, [
+        {
+          futureId: "Module1:Library1",
+          type: "execution-start",
+          futureType: FutureType.NAMED_LIBRARY_DEPLOYMENT,
+          strategy: "basic",
+          dependencies: [],
+          storedArtifactPath: "/user/path/Library1.json",
+          storedBuildInfoPath: "./build-info.json",
+          contractName: "Library1",
+          value: BigInt(0).toString(),
+          constructorArgs: [],
+          libraries: {},
+          from: accounts[0],
+        },
+        {
+          type: "onchain-action",
```

Comment:
I think I caught this quiet late, but shouldn't these messages have the futureId too?

Reply:
Yes, in fact we need to identify the future and the particular on-chain action.

They are added in the restart PR: https://github.com/NomicFoundation/ignition/blob/b61f9ac8a933db91d3dd937cec3bd25a3bdfe2c3/packages/core/src/new-api/types/journal.ts#L55

TransactionId is a bad name on reflection, maybe chainActionId would be better but we can discuss there.